### PR TITLE
adding checks that address i18n sync failures due to unexpected types

### PR DESCRIPTION
**Pull request by annaxuphoto**
In our most recent i18n sync, we encountered failures during the sync out regarding:
 - using a family name to retrieve a script instead of a script ID
 - attempting to operate on a String type as though it were a Hash

This PR adds checks to use the appropriate method for family names when appropriate, and to only symbolize keys for Hash objects.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by annaxuphoto** [comment]
This comment is created by running: `bin/i18n/analyze_pr.rb -p 45600 -c`

Top 2/2 files sorted by the number of changes:
- `dashboard/lib/services/i18n/curriculum_sync_utils/sync_out.rb`, 16 changes (9 additions, 7 deletions).
- `bin/i18n/i18n_script_utils.rb`, 6 changes (5 additions, 1 deletions).

Top 1/1 commits sorted by the number of changes:
- Commit: adding checks that address i18n sync failures due to unexpected types
  - 22 changes (14 additions, 8 deletions).
  - 2 files changed.
  - File with the most changes: `dashboard/lib/services/i18n/curriculum_sync_utils/sync_out.rb`, 16 changes (9 additions, 7 deletions).

File: bin/i18n/i18n_script_utils.rb [hunk 1]
```diff
@@ -168,7 +168,11 @@ def self.get_level_url_key(script, level)
 
   # Used by get_level_from_url, for the script_level-specific case.
   def self.get_script_level(route_params, url)
-    script = Script.get_from_cache(route_params[:script_id])
+    begin
+      script = Script.get_from_cache(route_params[:script_id])
+    rescue
+      script = Script.get_unit_family_redirect_for_user(route_params[:script_id])
```
Comment (daynew) [inline bin/i18n/i18n_script_utils.rb]: How did you figure out this was the way to handle it?

File: dashboard/lib/services/i18n/curriculum_sync_utils/sync_out.rb [hunk 2]
```diff
@@ -170,13 +170,15 @@ def self.postprocess(result)
         def self.flatten(object_hash, serializer, name)
           results = {}
           object_hash.each do |key, object|
-            object = object.symbolize_keys
+            if object.is_a?(Hash)
```
Comment (daynew) [inline dashboard/lib/services/i18n/curriculum_sync_utils/sync_out.rb]: Does this mean the content is in the wrong format?
Should we log a warning?
Should we instead handle the string case? `results[name][key] = object if object.is_a?(String)`

File: bin/i18n/i18n_script_utils.rb (hunk 1 above)
Comment (annaxuphoto) [reply bin/i18n/i18n_script_utils.rb]: super useful exception: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/script.rb#L591

File: dashboard/lib/services/i18n/curriculum_sync_utils/sync_out.rb (hunk 2 above)
Comment (annaxuphoto) [reply dashboard/lib/services/i18n/curriculum_sync_utils/sync_out.rb]: Good q! When I was investigating the issue, the String case I saw was a subset of a different object hash (e.g. one object was configured as `{'key': 'key', 'object': {'foo': 'bar'}}` before symbolizing keys, and the next was configured as `{'foo': 'bar'}`). Based on that example, I made the (maybe premature) assumption that the String case is redundant. I am on board with adding a warning though!

**Pull request closed without merging**